### ISSUE TITLE
[uni-uvpt] About build and pretrain with DCNv2

### ISSUE THREAD

**Issue by WJHash** [huawei-noah/noah-research#215]
Could you please kindly provide the versions of CUDA and PyTorch projects using? 
Project needs Deformable-Convolution-V2 (https://github.com/chengdazhi/Deformable-Convolution-V2-PyTorch.git) But it works on torch=0.4.1，and my environment is also unable to use the [DCNv2_pytorch_1.0.0 branch].

**Comment by WJHash** [comment]
The torch==1.10.2 in requirement.txt cant be used for DCNv2

**Comment by WJHash** [comment]
There also has no "MSDeformAttn" module in project Deformable-Convolution-V2

**Comment by WJHash** [comment]
I build it successfully with torch1.10.2，cuda 10.2，but get error
"python pretrain.py configs/pretrain/daformer_swin_base_patch4_window7_512x512_400k_2e-6_synthia.py"，when pretraining. After searching，I think there is still something wrong with env. Installing libtorch cant fix it

**Comment by WJHash** [comment]
I build it successfully with torch1.10.2，cuda 10.2，but get error
"python pretrain.py configs/pretrain/daformer_swin_base_patch4_window7_512x512_400k_2e-6_synthia.py"，when pretraining. After searching，I think there is still something wrong with env. Install libtorch cant fix it

**Comment by Siris-qy** [comment]
I’m facing the same issue. Have you managed to resolve it?

**Comment by WJHash** [comment]
> I’m facing the same issue. Have you managed to resolve it?

yes, use MultiScaleDeformableAttention from mmcv is ok for it. (MSDA is form DETR: https://github.com/fundamentalvision/Deformable-DETR/tree/main/models/ops)

**Comment by QChhh123** [comment]
@WJHash hello, I’m facing the same issue. Could you provide a conda list from the correct environment, thank you very much.
